sdk/nextjs/src/client/DevCycleClientsideProvider.tsx

```diff
@@ -0,0 +1,108 @@
+'use client'
+import React, { Suspense, use, useContext, useRef, useState } from 'react'
+import {
+    DevCycleClient,
+    DevCycleUser,
+    initializeDevCycle,
+} from '@devcycle/js-client-sdk'
+import { useRouter } from 'next/navigation'
+import { updateDVCCookie } from './updateDVCCookie'
+import { invalidateConfig } from '../common/invalidateConfig'
+import { DevCycleServerDataForClient } from '../common/types'
+
+type DevCycleClientsideProviderProps = {
+    serverDataPromise: Promise<DevCycleServerDataForClient>
+    sdkKey: string
+    user: DevCycleUser
+    enableStreaming: boolean
+    enableClientsideIdentify: boolean
+    children: React.ReactNode
+}
+
+type ClientProviderContext = {
+    client: DevCycleClient
+    sdkKey: string
+    enableStreaming: boolean
+    enableClientsideIdentify: boolean
+    serverDataPromise: Promise<unknown>
+}
+
+export const DevCycleClientContext = React.createContext<ClientProviderContext>(
+    {} as ClientProviderContext,
+)
+
+export const SuspendedProvider = ({
```

Comment:
this function could be one of the functions that have a little explanation on how it works generally